python/paddle/incubate/autograd/composite_rules.py

```diff
@@ -191,6 +191,35 @@ def mean_composite(x, axis, keepdim):
     return divide(sum_x, norm)
 
 
+@REGISTER_COMPOSITE('flatten_contiguous_range')
+def flatten_contiguous_range_composite(x, start_axis, stop_axis):
+    """
+    define composite rule of op flatten, flatten_contiguous_range -> flatten.
+    xshape is the dim with 0 added to the front of x, keep the shape information of x to calculate the grad.
+    shape_out is the parameter of reshape, get from start_axis and stop_axis.
+    out = reshape(x, shape=shape_out), xshape
+    """
+    shape_in = x.shape
+    shape_x_out = [0]
+    shape_x_out.extend(shape_in)
+    xshape = full(shape=shape_x_out, fill_value=0, dtype=x.dtype)
+    start_dim = start_axis if len(shape_in) != 0 else 0
+    end_dim = stop_axis if len(shape_in) != 0 else 0
+    assert start_dim <= end_dim
+    if len(shape_in) == 0 or start_dim == end_dim:
+        return reshape(x, shape=shape_in), xshape
+    slice_numel = 1
+    for i in range(start_dim, end_dim + 1):
+        slice_numel *= shape_in[i]
+    shape_out = []
+    for i in range(start_dim):
+        shape_out.append(shape_in[i])
+    shape_out.append(slice_numel)
+    for i in range(end_dim + 1, len(shape_in)):
+        shape_out.append(shape_in[i])
+    return reshape(x, shape=shape_out), xshape
```

Comment:
Can this op doesn't return xshape?

Reply:
There is xshape in the yaml definition, and need to use xshape when calculate grad.

Reply:
The yaml defineition :
https://github.com/PaddlePaddle/Paddle/blob/6786c012c144c29f976fc00e0e62f55dbb2d34ec/paddle/phi/api/yaml/legacy_ops.yaml#L644

Reply:
Yes, but for this case, which is a little special, CINN can't optimize xshape and actually it doesn't need xshape for backward pass. So, it's better to remove xshape usage if it's possible